com.unity.sg2/Editor/GraphUI/ShaderGraphGraphTool.cs

```diff
@@ -22,6 +44,119 @@ public override IToolbarProvider GetToolbarProvider()
+                        }
+                    }
+                    break;
+
+                // Handling value changes in ports
+                case UpdateConstantValueCommand updateConstantValueCommand:
+                    m_PreviewManager.HandleConstantValueUndoRedo(updateConstantValueCommand.Constant as BaseShaderGraphConstant);
+                    break;
+
+                case ChangeTargetSettingsCommand changeTargetSettingsCommand:
+                    m_PreviewManager.HandleTargetSettingsChanged();
+                    break;
+            }
+
+            m_PreviewManager.ReassignAllNodePreviewTextures();
+        }
+
+        internal void HandleGraphLoad(LoadGraphCommand loadGraphCommand)
+        {
+            var windowClosedInDirtyState = m_EditorWindow.wasWindowClosedInDirtyState;
+            BlackboardView blackboardView = null;
+            MainPreviewView mainPreviewView = null;
+
+            m_EditorWindow.TryGetOverlay(SGBlackboardOverlay.k_OverlayID, out var blackboard);
+            if (blackboard is SGBlackboardOverlay blackboardOverlay)
+                blackboardView = blackboardOverlay.BlackboardView;
+
+            m_EditorWindow.TryGetOverlay(PreviewOverlay.k_OverlayID, out var preview);
+            if (preview is PreviewOverlay previewOverlay)
+                mainPreviewView = previewOverlay.MainPreviewView;
+
+            var graphView = m_EditorWindow.GraphView;
+
+            var graphModelStateComponent = GetStateComponentOfType<GraphModelStateComponent>();
+            var graphViewStateComponent = GetStateComponentOfType<GraphViewStateComponent>();
+            var selectionStateComponent = GetStateComponentOfType<SelectionStateComponent>();
+            var undoStateComponent = GetStateComponentOfType<UndoStateComponent>();
+
+            var shaderGraphModel = loadGraphCommand.GraphModel as ShaderGraphModel;
+            // If handling a fresh load and not a reimport after a save
+            if (!wasGraphSaved)
+            {
+                m_PreviewManager.Initialize(
+                    graphModelStateComponent,
+                    shaderGraphModel,
+                    mainPreviewView,
+                    windowClosedInDirtyState);
+
+                ShaderGraphCommandsRegistrar.RegisterCommandHandlers(
+                    graphView,
+                    blackboardView,
+                    m_PreviewManager,
+                    shaderGraphModel,
+                    Dispatcher,
+                    graphModelStateComponent,
+                    graphViewStateComponent,
+                    selectionStateComponent,
+                    undoStateComponent);
+            }
+            else // if handling a graph reimport due to a save
+            {
+                m_PreviewManager.HandleGraphReload(graphModelStateComponent, shaderGraphModel, mainPreviewView);
+            }
+
+            wasGraphSaved = false;
+        }
+
```

Comment:
I want to dig deeper into why this is necessary. It seems like there is a collection of smaller changes in more local code locations that can accomplish the above without buying into additional CSO concepts.